templates/vdbench_save.yml

```diff
@@ -0,0 +1,160 @@
+
+- bash: |
+    export PATH=$PATH:${BUILD_SOURCESDIRECTORY}/test/utils
+
+    az network vnet subnet update \
+      --resource-group         ${RESOURCE_GROUP_NAME}-network \
+      --vnet-name              rendervnet \
+      --name                   render_clients1 \
+      --disable-private-endpoint-network-policies true
+    check_rc.sh $? "Disable private endpoint network policies on render_clients1 subnet" || exit 0
+
+    STOR_ACCT_RESOURCE_ID="/subscriptions/${ARM_SUBSCRIPTION_ID}/resourceGroups/${AZURE_STORAGE_ACCOUNT_RG_NAME}/providers/Microsoft.Storage/storageAccounts/${AZURE_STORAGE_ACCOUNT_NAME}"
+
+    az deployment group create \
+      -f ${BUILD_SOURCESDIRECTORY}/templates/private_endpoint.json \
+      --resource-group ${RESOURCE_GROUP_NAME}-network           \
+      -p location=${REGION}                                     \
+      -p privateEndpointName=${RESOURCE_GROUP_NAME}-privendpt   \
+      -p privateLinkResourceId=${STOR_ACCT_RESOURCE_ID}         \
+      -p subnet=render_clients1                                 \
+      -p vnet=rendervnet                                        \
+      -p subscriptionId=${ARM_SUBSCRIPTION_ID}
+    check_rc.sh $? "Create Private Endpoint" || exit 0
+  displayName: "Setup Private Endpoint for uploading vdbench results to Storage Account."
+  condition: not(or(failed(), canceled()))
+  env:
+    ARM_CLIENT_ID: $(AZURE-CLIENT-ID)
+    ARM_CLIENT_SECRET: $(AZURE-CLIENT-SECRET)
+    ARM_TENANT_ID: $(AZURE-TENANT-ID)
+    ARM_SUBSCRIPTION_ID: $(AZURE-SUBSCRIPTION-ID)
+    AZURE_STORAGE_ACCOUNT_NAME: $(AZURE-STORAGE-ACCOUNT-NAME)
+    AZURE_STORAGE_ACCOUNT_RG_NAME: $(AZURE-STORAGE-ACCOUNT-RG-NAME)
+
+- script: |
+    export PATH=$PATH:${BUILD_SOURCESDIRECTORY}/test/utils
+    end_timestamp=$(date -u -d "60 minutes" '+%Y-%m-%dT%H:%MZ')
+
+    dest_storacct_key=$(az storage account keys list            \
+        --account-name   ${AZURE_STORAGE_ACCOUNT_NAME}          \
+        --resource-group ${AZURE_STORAGE_ACCOUNT_RG_NAME}       \
+        --subscription   $ARM_SUBSCRIPTION_ID --query "[1].value" --output tsv)
+    rc=$?
+    echo "##vso[task.setvariable variable=dest_storacct_key;issecret=true]${dest_storacct_key}"
+    check_rc.sh $rc "Get destination Storage Account key" || exit 0
+
+    dest_container_sas=$(az storage container generate-sas      \
+        --account-key  "${dest_storacct_key}"                   \
+        --account-name ${AZURE_STORAGE_ACCOUNT_NAME}            \
+        --name         ${AZURE_STORAGE_ACCOUNT_CONTAINER_NAME}  \
+        --expiry       "${end_timestamp}"                       \
+        --permissions clrw --https-only --output tsv)
+    rc=$?
+    echo "##vso[task.setvariable variable=dest_container_sas;issecret=true]${dest_container_sas}"
+    check_rc.sh $rc "Generate destination SAS URL" || exit 0
+
+    sas_url="https://${AZURE_STORAGE_ACCOUNT_NAME}.blob.core.windows.net/${AZURE_STORAGE_ACCOUNT_CONTAINER_NAME}?${dest_container_sas}"
+    rc=$?
```

Comment:
This is just a string interpolation in bash and is unlikely to fail. Probably don't need to store `$rc`. Maybe a holdover from the `$(echo ...)` iteration.

Reply:
removed